engine/src/history.rs

```diff
@@ -34,8 +34,8 @@ impl History {
         his
     }
 
-    pub fn record_move(&mut self, piece: String, pos: String) {
-        self.moves.push((piece, pos));
+    pub fn record_move(&mut self, piece: &str, pos: &str) {
+        self.moves.push((piece.to_string(), pos.to_string()));
```

Comment:
why not store the history as structs, or at least `(Piece, Position)`s?

Reply:
The history is a struct but one of the members being `moves` :)

Reply:
right, i mean why not have `moves` be something like `Vec<Turn>`?

Reply:
Will address this in a separate PR :)